libr/asm/p/asm_mips_gnu.c

```diff
@@ -43,8 +45,36 @@ static int disassemble(struct r_asm_t *a, struct r_asm_op_t *op, const ut8 *buf,
 	Offset = a->pc;
 	memcpy (bytes, buf, 4); // TODO handle thumb
 
+	if ((a->cpu != pre_cpu) && (a->features != pre_features)) {
+		memset (&disasm_obj,'\0', sizeof (struct disassemble_info));
```

Comment:
when resetting the structure `disasm_obj.disassembler_options` should be freed as well, as it contains a reference to a string allocated with `r_str_new`

Reply:
This is a static variable defined in the previous code, I just want to adapt it

Reply:
My bad